zkevm-circuits/src/evm_circuit.rs

```diff
@@ -1 +1,916 @@
+        );
+
+        (
+            qs_op_execution,
+            qs_byte_lookups,
+            op_execution_state_curr,
+            op_execution_state_next,
+            op_execution_free_cells,
+        )
+    }
+
+    fn configure_lookups(
+        meta: &mut ConstraintSystem<F>,
+        qs_byte_lookup: Column<Advice>,
+        advices: [Column<Advice>; CIRCUIT_WIDTH],
+        fixed_table: [Column<Fixed>; 4],
+        rw_table: [Column<Advice>; 7],
+        op_execution_state_curr: OpExecutionState<F>,
+        independent_lookups: Vec<(Expression<F>, Vec<Lookup<F>>)>,
+    ) {
+        // TODO: call_lookups
+        // TODO: bytecode_lookups
+
+        let mut fixed_lookups = Vec::<[Expression<F>; 4]>::new();
+        let mut rw_lookups = Vec::<[Expression<F>; 7]>::new();
+
+        for (qs_lookup, lookups) in independent_lookups {
+            let mut fixed_lookup_count = 0;
+            let mut rw_lookup_count = 0;
+
+            for lookup in lookups {
+                match lookup {
+                    Lookup::FixedLookup(tag, exprs) => {
+                        let exprs = iter::once(tag.expr()).chain(exprs.clone());
+
+                        if fixed_lookups.len() == fixed_lookup_count {
+                            fixed_lookups.push(
+                                exprs
+                                    .map(|expr| qs_lookup.clone() * expr)
+                                    .collect::<Vec<_>>()
+                                    .try_into()
+                                    .unwrap(),
+                            );
+                        } else {
+                            for (acc, expr) in fixed_lookups[fixed_lookup_count]
+                                .iter_mut()
+                                .zip(exprs)
+                            {
+                                *acc = acc.clone() + qs_lookup.clone() * expr;
+                            }
+                        }
+                        fixed_lookup_count += 1;
+                    }
+                    Lookup::BusMappingLookup(
+                        rw_lookup
+                        @
+                        (BusMappingLookup::Stack { .. }
+                        | BusMappingLookup::Memory { .. }
+                        | BusMappingLookup::AccountStorage {
+                            ..
+                        }),
+                    ) => {
+                        let OpExecutionState {
+                            global_counter,
+                            call_id,
+                            stack_pointer,
+                            ..
+                        } = &op_execution_state_curr;
+
+                        if rw_lookups.len() == rw_lookup_count {
+                            rw_lookups
+                                .push(vec![0.expr(); 7].try_into().unwrap());
+                        }
+
+                        let mut exprs = vec![
+                            global_counter.expr() + rw_lookup_count.expr(),
```

Comment:
not sure why need to plus rw_lookup_count.expr() ?  if so , even the same buss mapping lookup will make different exprs .

Reply:
We might have several `rw_lookup` in single step, but since global counter serves as the order of access, we should ensure its uniqueness property. Otherwise we have no way to check if a malicious prover skips or inserts some write in bus mapping to cheat.

> even the same bus mapping lookups will make different exprs.

Why does it have same lookups in single step?

Reply:
I tested same lookups in single step  condition, so be aware of it.  avoiding same lookups in single step seems make sense.